### PR TITLE
Feat(eos_cli_config_gen): Add schema for mac_security

### PR DESCRIPTION
**Pull request by ghost**
### Contributor Checklist

- [x] Create schema fragment matching data model described in README.md and README_v4.0.md
  - README.md is most complete with all keys. README_v4.0 includes partial data models after conversion to lists.
  - Schema fragment path is `roles/eos_cli_config_gen/schemas/schema_fragments/<data_model_key>.schema.yml`.
  - Copy line 1-5 from another schema (comments and outer type:dict).
  - Refer to [schema documentation](https://avd.sh/en/devel/docs/input-variable-validation-BETA.html) for syntax and/or use YAML Lint plugin from Redhat in VSCode.
  - Use `convert_types` on value that could be mixed type or misinterpreted like integers and numeric strings.
- [x] If the data model has been converted from wildcard dicts:
  - Add `convert_types: ['dict']` to the schema.
  - Remove `convert_dicts` from the `templates/eos/<>.j2` and `templates/documentation/<>.j2` templates.
- [x] Run `molecule converge -s build_schemas_and_docs` to update schema and documentation.
- [x] Test by running `molecule converge -s eos_cli_config_gen` and verify no errors or changes to generated configs/docs.

### Reviewer Checklist

- Reviewer 1: Carl
  - [x] Verify that data model is fully covered in the described schema. Easiest by looking at the generated documentation.
  - [x] Verify that `convert_dicts` has been removed from templates as applicable.
  - [x] Verify no changes to configs/docs on any molecule scenario
  - [x] Verify that CI pass

- Reviewer 2: Emil
  - [x] Verify that data model is fully covered in the described schema. Easiest by looking at the generated documentation.
  - [x] Verify that `convert_dicts` has been removed from templates as applicable.
  - [x] Verify no changes to configs/docs on any molecule scenario
  - [x] Verify that CI pass

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/eos_cli_config_gen.schema.yml [hunk 1]
```diff
@@ -332,6 +332,68 @@ keys:
...
+    keys:
+      license:
+        type: dict
+        required: true
+        keys:
+          license_name:
+            type: str
+            required: true
+          license_key:
+            type: int
```
Comment (emilarista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/eos_cli_config_gen.schema.yml]: This should be str type not int. Also won't need the convert_types after this change.

Comment (carlbuchmann) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/eos_cli_config_gen.schema.yml]: addressed in commit [57ecdf2](https://github.com/aristanetworks/ansible-avd/pull/2058/commits/57ecdf2f5950dad1e183c198de33926616fcf070)

**Comment by Shivani-chourasiya** [comment]
Thanks @carlbuchmann

**Review by emilarista — APPROVED** [review]
LGTM!

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/mac_security.schema.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        type: dict
+        required: true
+        keys:
+          license_name:
+            type: str
+            required: true
+          license_key:
+            type: str
+            required: true
+            display_name: License-Number
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/mac_security.schema.yml]:
```suggestion
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/mac_security.schema.yml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        items:
+          type: dict
+          keys:
+            name:
+              type: str
+              required: true
+              display_name: Profile-Name
+            cipher:
+              type: str
+              description: valid-cipher-string
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/mac_security.schema.yml]:
```suggestion
              valid_values:
                - aes128-gcm
                - aes128-gcm-xpn
                - aes256-gcm
                - aes256-gcm-xpn
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/mac_security.schema.yml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+              type: dict
+              keys:
+                session:
+                  type: dict
+                  keys:
+                    rekey_period:
+                      type: int
+                      convert_types:
+                        - str
+                      description: 30-100000 in seconds
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/mac_security.schema.yml]:
```suggestion
                      min: 30
                      max: 100000
                      description: Rekey period in seconds
```

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

**Comment by ClausHolbechArista** [comment]
@carlbuchmann please review this

**Review by ClausHolbechArista — APPROVED** [review]

**Review by carlbuchmann — APPROVED** [review]
LGTM!

**Pull request merged**